lib/internal/streams/writable.js

```diff
@@ -636,7 +641,9 @@ function onwrite(stream, er) {
         state.state |= (kAfterWritePending | kAfterWriteTickInfo);
       } else {
         state.pendingcb--;
-        finishMaybe(stream, state, true);
+        if ((state.state & kEnding) !== 0) {
```

Comment:
We may as well optimize needFinish?

Reply:
I think this is more about inlining.

Reply:
We could maybe inline needFinish but that would impact readability for little gain